src/common/mod.rs

```diff
@@ -31,6 +31,9 @@ pub const MEBIBYTE: u64 = 1_024 * KIBIBYTE;
 pub const KIBIBYTE: u64 = 1_024 * BYTE;
 pub const BYTE: u64 = 1;
 
+#[allow(dead_code)]
+pub const SAMPLE_PERIOD: u64 = 1;
```

Comment:
Have we profiled the overhead here? I still would think a frequency based approach is better.